src/transport/raw/BLE.cpp

```diff
@@ -66,8 +66,16 @@ CHIP_ERROR BLEBase::Init(const BleListenParameters & param)
     VerifyOrReturnError(mState == State::kNotReady, CHIP_ERROR_INCORRECT_STATE);
     VerifyOrReturnError(bleLayer != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
-    mBleLayer                           = bleLayer;
-    mBleLayer->mBleTransport            = this;
+    mBleLayer = bleLayer;
+    if (mBleLayer->mBleTransport == nullptr || param.HasOverrideExistingTransport())
+    {
+        mBleLayer->mBleTransport = this;
```

Comment:
please also update ClearState so that it clears mBleLayer->mBleTransport. I think leaving the dangling pointer there is not correct.